### PR TITLE
Rework envproxy and cemanager environment variables for RPs

### PR DESCRIPTION
**Pull request by deansheather**
Reworks env vars to match the sister PR #17. That contains a fuller list of changes.

### CODE REVIEW

File: templates/cemanager.yaml
Comment (f0ssel) [inline templates/cemanager.yaml:42]: So envproxy is now the sole owner of the service account (that does anything), as in that service account will be the separate envproxy helm chart?

File: templates/envproxy.yaml
Comment (f0ssel) [inline templates/envproxy.yaml:33]: This means envproxy itself can also easily make kube calls. There could be a future where envproxy runs the controllers and sends build events back to cemanager. (this idea was in the OG resource pools rfc I think) to keep the kube data local and move the responsibility out of cemanager altogether.

**Review by f0ssel — APPROVED** [review]
LGTM, When does this get merged in relation to the others stuff?

File: templates/envproxy.yaml
Comment (deansheather) [reply templates/envproxy.yaml:33]: I had to do this because it's difficult to mount a second service account unless you know the secret's name exactly AFAIK.

File: templates/cemanager.yaml
Comment (deansheather) [reply templates/cemanager.yaml:42]: This will be the same in the envproxy chart. Envproxy will assume the `coder` SA, and forward that to the cemanager.

**Comment by deansheather** [comment]
I'll test this in my branch by checking out this submodule branch in `m`, after you merge your branch into master in `m` and I rebase on that.

File: templates/envproxy.yaml
Comment (f0ssel) [reply templates/envproxy.yaml:33]: no I think this is better :+1:

**Pull request merged**
